### PR TITLE
A handful of tentative performance improvements

### PR DESCRIPTION
**Pull request by krtab**
@zapashcanon let me know if you want them in separate PRs

### CODE REVIEW

File: src/symbolic/symbolic_choice.mli [hunk 1]
```diff
@@ -13,4 +13,11 @@ include
...
      and module V := Symbolic_value
 
-val run : workers:int -> 'a t -> Thread.t -> 'a run_result
+val run :
+     workers:int
+  -> 'a t
+  -> Thread.t
+  -> ('b -> unit)
+  -> (unit -> unit)
+  -> (unit -> unit)
```
Comment (zapashcanon) [inline src/symbolic/symbolic_choice.mli:22]: Could you use named parameters for the three parameters ?

File: src/cmd/cmd_sym.ml [hunk 2]
```diff
@@ -244,7 +244,21 @@ let cmd profiling debug unsafe optimize workers no_stop_at_failure no_values
   let pc = Choice.return (Ok ()) in
   let result = List.fold_left (run_file ~unsafe ~optimize) pc files in
   let thread : Thread.t = Thread.create () in
-  let results = Symbolic_choice.run ~workers result thread in
+  let res_queue = Wq.init () in
+  let callback v =
+    match fst v with
+    | Symbolic_choice.EVal (Ok ()) -> ()
+    | _ -> Wq.push v res_queue
```
Comment (zapashcanon) [inline src/cmd/cmd_sym.ml:251]:
```ocaml
let callback = function
  | Symbolic_choice.EVal (Ok ()), _ -> ()
  | _ -> Wq.push v res_queue
```

File: dune-project [hunk 3]
```diff
@@ -74,5 +74,6 @@
   digestif
   xmlm
   (hc (>= 0.3))
+  (processor (= 0.1))
```
Comment (zapashcanon) [inline dune-project:77]: `>= 0.1`

File: src/bin/owi.ml [hunk 4]
```diff
@@ -65,7 +65,7 @@ let workers =
   in
```
Comment (zapashcanon) [inline src/bin/owi.ml:65]: Could you update the doc to say we use the total number of physical cores and not the value provided by Domain.recommended_domain_count anymore ?

File: src/symbolic/symbolic_choice.ml [hunk 5]
```diff
@@ -366,7 +366,7 @@ let get_model_or_stop symbol =
     | Some v -> return v
   end
 
-let select (cond : Symbolic_value.vbool) =
+let select_inner ?(explore_first = true) (cond : Symbolic_value.vbool) =
```
Comment (zapashcanon) [inline src/symbolic/symbolic_choice.ml:369]: Could you use `~explore_first` instead ? I don't really like optional arguments TBH, and here the function is used in only two places so it's not that tedious

**Comment by zapashcanon** [comment]
LGTM once we checked that the 1st and 3rd one are actually leading to better results, which I would expect (+ my nitpicks comments).

**Comment by zapashcanon** [comment]
@krtab, I wanted to run the benchmarks on the server, but the code actually does not typecheck

**Comment by krtab** [comment]
Split in #320 #321 #322 #323

**Pull request closed without merging**